tests/canary_upgrade_test.go

```diff
@@ -99,24 +98,23 @@ var _ = Describe("[Serial][sig-operator]virt-handler canary upgrade", Serial, de
 	}
 
 	updateVirtHandler := func() error {
-		kv := util.GetCurrentKv(virtCli)
-
-		patch := fmt.Sprintf(`{"spec": { "template": {"metadata": {"annotations": {"%s": "test"}}}}}`,
+		testPatch := fmt.Sprintf(`{"spec": { "template": {"metadata": {"annotations": {"%s": "test"}}}}}`,
 			e2eCanaryTestAnnotation)
-		kv.Spec.CustomizeComponents = v1.CustomizeComponents{
+
+		ccs := v1.CustomizeComponents{
 			Patches: []v1.CustomizeComponentsPatch{
 				{
 					ResourceName: "virt-handler",
 					ResourceType: "DaemonSet",
 					Type:         v1.StrategicMergePatchType,
-					Patch:        patch,
+					Patch:        testPatch,
 				},
 			},
 		}
-		return retry.RetryOnConflict(retry.DefaultRetry, func() error {
-			_, err := virtCli.KubeVirt(flags.KubeVirtInstallNamespace).Update(context.Background(), kv, metav1.UpdateOptions{})
-			return err
-		})
+
+		patchPayload, err := patch.New(patch.WithReplace("/spec/customizeComponents", ccs)).GeneratePayload()
```

Comment:
You should return in case of an error.
Currently, the `err` variable is not valued and it is overwritten by the `Patch` performed later.

Reply:
not sure how i missed that, thanks